### PR TITLE
COMMERCE-9805 - It's possible to insert via API 0 or negative numeric values on product subscription fields

### PR DESCRIPTION
**Pull request by victorcdp**
Related Issue: [COMMERCE-9805](https://issues.liferay.com/browse/COMMERCE-9805)
Resent from #3315. This time the only fix needed is for the Subscription Cycles, since the other values were already fixed by [COMMERCE-10831](https://issues.liferay.com/browse/COMMERCE-10831)

The issue was that there was no backend validation for the input numbers.
Proposed fix is to add validation at the addCPDefinition and updateSubscriptionInfo methods.

### CODE REVIEW

**Comment by victorcdp** [comment]
ci:test:sf

**Comment by victorcdp** [comment]
ci:test:relevant

**Comment by victorcdp** [comment]
ci:test:commerce

**Comment by victorcdp** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java [hunk 1]
```diff
@@ -3201,6 +3210,18 @@ private void _validate(
 		return null;
 	}
 
+	private void _validateSubscriptionCycles(
+			long subscriptionCycles, String subscriptionCyclesKey)
+		throws PortalException {
+
+		if (subscriptionCycles < 0) {
+			throw new CPDefinitionMaxSubscriptionCyclesException(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java:3218]: please throw a CPDefinitionDeliveryMaxSubscriptionCycles for deliverySubscriptionCycles case

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/jaxrs/exception/mapper/DefinitionDeliveryMaxSubscriptionCyclesExceptionMapper.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	extends BaseExceptionMapper
+		<CPDefinitionDeliveryMaxSubscriptionCyclesException> {
+
+	@Override
+	protected Problem getProblem(
+		CPDefinitionDeliveryMaxSubscriptionCyclesException
+			cpDefinitionDeliveryMaxSubscriptionCyclesException) {
+
+		return new Problem(
+			Response.Status.BAD_REQUEST,
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/jaxrs/exception/mapper/DefinitionDeliveryMaxSubscriptionCyclesExceptionMapper.java:47]: @alessio-rendina is this right according to the new pattern?

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java (hunk 1 above)
Comment (victorcdp) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java:3221]: Changed it, should be good now. Thank you

**Review by andrea-ale-sbarra — APPROVED** [review]
LGTM, thank you!

The only doubt that I have is about the ExceptionMapper, we introduced some new pattern to make EM, and I'm not 100% this is following it.
@alessiorendina can you please check it?

Thank you!

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/jaxrs/exception/mapper/DefinitionDeliveryMaxSubscriptionCyclesExceptionMapper.java (hunk 2 above)
Comment (alessiorendina) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/jaxrs/exception/mapper/DefinitionDeliveryMaxSubscriptionCyclesExceptionMapper.java:47]: Yes, for now it's good. We need to refactor all of them in the future.

**Review by alessiorendina — APPROVED** [review]

**Comment by Brunno0liveira** [comment]
ci:forward

**Comment by Brunno0liveira** [comment]
ci:stop

**Comment by Brunno0liveira** [comment]
ci:stop:forward

**Comment by Brunno0liveira** [comment]
ci:stop

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by victorcdp** [comment]
ci:test:sf

**Comment by victorcdp** [comment]
ci:test:relevant

**Comment by victorcdp** [comment]
ci:test:commerce

**Comment by Brunno0liveira** [comment]
ci:test:commerce-functional

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:relevant

**Comment by Brunno0liveira** [comment]
ci:forward

**Pull request closed without merging**